### DISCUSSION TITLE
Windows build instructions

### DISCUSSION THREAD

**Discussion by lampra1** [SBFspot/SBFspot#624] [Q&A] [answered]
Hi, Could you provide instructions to build the Windows sqlite version?
I want to test the settime branch for a site with multiple inverters.
Thank you

**Comment by sbf-** [comment]
It's a bit complicated... I have my build environment, but it's undocumented.
I can make a windows executable of the `settime branch` if you like

**Reply by lampra1** [reply]
> I can make a windows executable of the `settime branch` if you like

Yes please do so, thank you

**Reply by sbf-** [reply, marked as answer]
Here it is:
[SBFspot_V382_settime_W32_SQLite.zip](https://github.com/SBFspot/SBFspot/files/11160714/SBFspot_V382_settime_W32_SQLite.zip)

**Reply by lampra1** [reply]
Thank you, I will give it a try for a few days and I will get back to you.

**Comment by lampra1** [comment]
It seems that this did not solve the issue I am facing. 
The issue is that two inverters "loose" time every evening, and they revert to 2001.
The SMA support said that the inverters need to be repaired but the cost is rather high.

The only workaround that seems to work is to turn back the pc time by 15 seconds, set the inverters time, and then reset the time of the pc. I do this using task scheduler and the latest sbfspot tag (3.9.3) and PowerShell (7.3.4) 3 times (every hour) in the morning to ensure that time is set. 
The command for anyone interested is:
`pwsh.exe -Command (Get-Date).AddSeconds(-15) | Set-Date |  SBFspot.exe -v -settime -finq || (Get-Date).AddSeconds(15) | Set-Date`

**Reply by sbf-** [reply]
> It seems that this did not solve the issue I am facing

Looks like `SBFspot -settime` does its job as it sets the inverter time.

> The SMA support said that the inverters need to be repaired but the cost is rather high

Just a thought: there might be a backup battery like CR2032 to keep time of the RTC clock when the inverter is switched off. Every PC has one, so maybe this is the case for SMA inverters?

**Comment by lampra1** [comment]
> Looks like SBFspot -settime does its job as it sets the inverter time.

Indeed, but I was hopping to do that without having to revert the pc clock every time

> Just a thought: there might be a backup battery like CR2032 to keep time of the RTC clock when the inverter is switched off. Every PC has one, so maybe this is the case for SMA inverters?

IIRW, old SMA inverters don't have such a battery, but I will have a look
